### PR TITLE
Added better lighting for Streetlight.

### PR DESCRIPTION
**Pull request by NathanSalapat**
Walkable = false lets people walk through the node, more aptly put it prevents people from walking ontop of the node.
I dropped the light source of the streetlight down to 4, and added the lamp light at 14.
Selection and Collision box definitions were moved above the groups and callbacks.
Added the on_construct and on_destruct callbacks to add and remove the invisible light node.
Added the invisible light node.

### CODE REVIEW

**Comment by Steamed-Punk** [comment]
That was fast. It looks so much better, natural. Exactly what I was looking for.
As before I will make the necessary credit changes to files and link back to you.
Thank you "**Once again**" for your time.

**Comment by NathanSalapat** [comment]
I figured it would just be easier to make a pull request then try and explain what you needed to change in the code so you could do it yourself.

**Comment by Steamed-Punk** [comment]
You are right, even if you did explain in layman's terms I am sure you would end up being more frustrated at me than I was with the code.
